compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -421,13 +421,17 @@ pub trait Machine<'mir, 'tcx: 'mir>: Sized {
     /// Called on places used for in-place function argument and return value handling.
     ///
     /// These places need to be protected to make sure the program cannot tell whether the
-    /// argument/return value was actually copied or passed in-place..
+    /// argument/return value was actually copied or passed in-place.
     fn protect_in_place_function_argument(
         ecx: &mut InterpCx<'mir, 'tcx, Self>,
         place: &PlaceTy<'tcx, Self::Provenance>,
     ) -> InterpResult<'tcx> {
         // Without an aliasing model, all we can do is put `Uninit` into the place.
-        ecx.write_uninit(place)
+        // This can only be violated with custom MIR though so we avoid the perf hit.
+        if ecx.tcx.sess.opts.unstable_opts.extra_const_ub_checks {
```

Comment:
@RalfJung , this flag has to be turned on explicitly, right?

Can/should we also unconditionally apply this logic when the code being compiled was defined via `#[custom_mir(..)]` ?

Reply:
Indeed this is an off-by-default flag, also guarding whether we check for value validity on every typed copy.

I feel like even with custom MIR, causing UB here is so unlikely that it's not worth the trouble. However, for some reason it seems that just checking this flag is already sufficient to mostly negate all perf benefits? Or the perf changes we saw earlier were mostly noise.